storage/internal/benchmarks/client_pool.go

```diff
@@ -132,81 +136,82 @@ func initializeClientPools(ctx context.Context, opts *benchmarkOptions) func() {
 
-func initializeHTTPClient(ctx context.Context, p clientParams) (*storage.Client, error) {
-	if !p.customClient {
-		clientMu.Lock()
-		c, err := storage.NewClient(ctx)
-		clientMu.Unlock()
-		return c, err
-	}
+	if writeBufferSize != useDefault || readBufferSize != useDefault {
+		// We need to modify the underlying HTTP client
+		dialer := &net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}
 
-	dialer := &net.Dialer{
-		Timeout:   30 * time.Second,
-		KeepAlive: 30 * time.Second,
-	}
+		// These are the default parameters with write and read buffer sizes modified
+		base := &http.Transport{
+			Proxy:                 http.ProxyFromEnvironment,
+			DialContext:           dialer.DialContext,
+			ForceAttemptHTTP2:     true,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+			WriteBufferSize:       writeBufferSize,
+			ReadBufferSize:        readBufferSize,
+		}
 
-	// These are the default parameters with write and read buffer sizes modified
-	base := &http.Transport{
-		Proxy:                 http.ProxyFromEnvironment,
-		DialContext:           dialer.DialContext,
-		ForceAttemptHTTP2:     true,
-		MaxIdleConns:          100,
-		IdleConnTimeout:       90 * time.Second,
-		TLSHandshakeTimeout:   10 * time.Second,
-		ExpectContinueTimeout: 1 * time.Second,
-		WriteBufferSize:       p.writeBufferSize,
-		ReadBufferSize:        p.readBufferSize,
-	}
+		http2Trans, err := http2.ConfigureTransports(base)
+		if err == nil {
+			http2Trans.ReadIdleTimeout = time.Second * 31
+		}
 
-	http2Trans, err := http2.ConfigureTransports(base)
-	if err == nil {
-		http2Trans.ReadIdleTimeout = time.Second * 31
+		trans, err := htransport.NewTransport(ctx, base,
+			option.WithScopes("https://www.googleapis.com/auth/devstorage.full_control"))
+		if err != nil {
+			return nil, err
+		}
+
+		opts = append(opts, option.WithHTTPClient(&http.Client{Transport: trans}))
 	}
 
-	trans, err := htransport.NewTransport(ctx, base,
-		option.WithScopes("https://www.googleapis.com/auth/devstorage.full_control"))
-	if err != nil {
-		return nil, err
+	if json {
+		opts = append(opts, storage.WithJSONReads())
 	}
 
+	// Init client
 	clientMu.Lock()
-	client, err := storage.NewClient(ctx, option.WithHTTPClient(&http.Client{Transport: trans}))
+	client, err := storage.NewClient(ctx, opts...)
 	clientMu.Unlock()
 
 	return client, err
 }
 
-func initializeGRPCClient(ctx context.Context, p clientParams) (*storage.Client, error) {
-	if !p.customClient {
-		clientMu.Lock()
-		os.Setenv("STORAGE_USE_GRPC", "true")
-		c, err := storage.NewClient(ctx)
-		os.Unsetenv("STORAGE_USE_GRPC")
-		clientMu.Unlock()
-		return c, err
+func initializeGRPCClient(ctx context.Context, writeBufferSize, readBufferSize int, connectionPoolSize int) (*storage.Client, error) {
+	opts := []option.ClientOption{option.WithGRPCConnectionPool(connectionPoolSize)}
```

Comment:
Write/read buffers are conditional due to the need to modify the underlying HTTP client and since they have different default values for GRPC and HTTP. 

For connection pool size, the default is set to 4 on receiving the param so we can just always add this option as it really only sets a value and would be setting it to the default if nothing is provided. I could wire this param similar to read/write buffer params if there is a good case for it though.

Reply:
I would just make it conditional as well in case we change the default conn pool size in the client in the future.

Reply:
If it does change, then we don't have a way of telling in the results. ie. it will still print `4` even though the default changed and now it's `6`. Is that acceptable?

Reply:
Hmm I guess we generally have this problem with default values. Do we always specify ChunkSize for example? I guess either approach is fine as long as we are consistent.

Reply:
Chunksize is specified as we can check the default value for it in code (open a writer and get the chunksize). For values that we cannot, we have these options:

a - assign the value as the current default. The output would have this value. If the default changes, we would need to update the code.
b - make the value conditional. The output would have a value like "-1" to indicate default. Personally I don't love this since the output looks incomplete but we had been doing that at some point previously; it's easy and it will always "do the right thing".
c - make the value conditional but set the value to the current default in the output. If the value changes, the output would be lying.

Another option might be using reflection to try and obtain the value from the client.

Reply:
Whatever we decide, I think I will leave that for another PR so we can more easily see what's changing.